plugins/lookup/cluster_moid.py

```diff
@@ -11,33 +11,102 @@
 name: cluster_moid
 short_description: Look up MoID for vSphere cluster objects using vCenter REST API
 description:
-    - Returns Managed Object Reference (MoID) of the vSphere cluster object contained in the specified path.
+  - Returns Managed Object Reference (MoID) of the vSphere cluster object contained in the specified path.
 author:
-    - Alina Buzachis (@alinabuzachis)
+  - Alina Buzachis (@alinabuzachis)
 version_added: 2.1.0
 requirements:
-    - vSphere 7.0.3 or greater
-    - python >= 3.6
-    - aiohttp
+  - vSphere 7.0.3 or greater
+  - python >= 3.6
+  - aiohttp
 extends_documentation_fragment:
-- vmware.vmware_rest.moid
+  - vmware.vmware_rest.moid
 """
 
 
 EXAMPLES = r"""
-# lookup sample
-- name: set connection info
+#
+#
+# The examples below assume you have a datacenter named 'my_dc' and a cluster named 'my_cluster'.
+# Replace these values as needed for your environment.
+#
+#
+
+#
+# Authentication / Connection Arguments
+#
+# You can explicitly set the connection arguments in each lookup. This may be clearer for some use cases
+- name: Pass In Connection Arguments Explicitly
+  ansible.builtin.debug:
+    msg: >-
+      {{ lookup('vmware.vmware_rest.cluster_moid', '/my_dc/host/my_cluster',
+      vcenter_hostname="vcenter.test",
+      vcenter_username="administrator@vsphere.local",
+      vcenter_password="1234") }}
+
+# Alternatively, you can add the connection arguments to a dictionary variable, and then pass that variable to the
+# lookup plugins. This makes the individual lookup plugin calls simpler
+- name: Set Connection Argument Variable
   ansible.builtin.set_fact:
```

Comment:
ah, thats a good tip. I am not sure a block makes sense here though, since the example is showing the user can set this auth variable once early in the playbook and then reference it throughout the play. 
I could update the example to be a stub playbook though? like
```
- hosts: all
  vars:
    connection_args:
      .....
  tasks:
    - debug: ...
```